packages/flutter/lib/src/material/material_app.dart

```diff
@@ -88,6 +95,15 @@ class _MaterialAppState extends State<MaterialApp> implements BindingObserver {
 
   void didChangeSize(Size size) => setState(() { _size = size; });
 
+  void didChangeLocale(ui.Locale locale) => _updateLocaleState(locale);
+
+  void _updateLocaleState(ui.Locale locale) {
+    if (config.onLocaleChanged != null) {
+      config.onLocaleChanged(locale).then(
+          (LocaleQueryData data) => setState(() { _localeData = data; }));
```

Comment:
nit: our style guide says to have the same whitespace before the ) and } as after the corresponding ( and {, to avoid having indents that don't have matching closes.